src/hotspot/share/opto/gcm.cpp

```diff
@@ -669,6 +669,20 @@ Block* PhaseCFG::insert_anti_dependences(Block* LCA, Node* load, bool verify) {
   // The anti-dependence constraints apply only to the fringe of this tree.
 
   Node* initial_mem = load->in(MemNode::Memory);
+  // We don't optimize the memory graph for pinned loads, so we may need to raise the
+  // root of our search tree through the corresponding slices of MergeMem nodes to
+  // get to the node that really creates the memory state for this slice.
+  if (load_alias_idx >= Compile::AliasIdxRaw) {
+    while (initial_mem->is_MergeMem()) {
+      MergeMemNode* mm = initial_mem->as_MergeMem();
+      Node* p = mm->memory_at(load_alias_idx);
+      if (p != mm->base_memory()) {
+        initial_mem = p;
+      } else {
+        break;
+      }
+    }
+  }
   worklist_store.push(initial_mem);
   worklist_visited.push(initial_mem);
   worklist_mem.push(NULL);
```

Comment:
OK. The JDK 17u version has the `nullptr` vs `NULL` changes. Hence the patch is not deemed clean.

Reply:
I see, thank you. 
Is there anything expected from me here (like removing "This is a clean backport." from the original comment) or is it just for me to know?